### PR TITLE
[ISSUE-818][REFACTOR] Move existing RssConf.xxx conf method to RssConf class

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### Related issues.
close https://github.com/alibaba/celeborn/issues/818

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Please adding verb "fix" or "close" before the issue ref `#<number>` in your PR description, so that GitHub can auto close the issue when PR got merged.

**Comment by AngersZhuuuu** [comment]
> ### What changes were proposed in this pull request?
> ### Why are the changes needed?
> ### What are the items that need reviewer attention?
> ### Related issues.
> #818
> 
> ### Related pull requests.
> ### How was this patch tested?
> /cc @related-reviewer
> 
> /assign @main-reviewer



> Please add the verb "fix" or "close" before the issue ref `#<number>` in your PR description, so that GitHub can auto-close the issue when PR got merged.

Like this? cc @nafiyAix

**Comment by pan3793** [comment]
yep

File: client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ public SortBasedPusher(
             afterPush,
             mapStatusLengths);
 
-    pushBufferSize = RssConf.pushBufferMaxSize(conf);
+    pushBufferSize = conf.pushBufferMaxSize();
```
Comment (FMX) [inline client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java:113]: The variable should be renamed too.

File: client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/HashBasedShuffleWriter.java [hunk 2]
```diff
@@ -130,8 +130,8 @@ public HashBasedShuffleWriter(
     mapStatusRecords = new long[numPartitions];
     tmpRecords = new long[numPartitions];
 
-    SEND_BUFFER_INIT_SIZE = RssConf.pushBufferInitialSize(conf);
-    SEND_BUFFER_SIZE = RssConf.pushBufferMaxSize(conf);
+    SEND_BUFFER_INIT_SIZE = conf.pushBufferInitialSize();
```
Comment (FMX) [inline client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/HashBasedShuffleWriter.java:133]: ditto.

File: client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java [hunk 3]
```diff
@@ -118,7 +118,7 @@ public SortBasedShuffleWriter(
     this.mapStatusRecords = new long[numPartitions];
     tmpRecords = new long[numPartitions];
 
-    pushBufferSize = RssConf.pushBufferMaxSize(conf);
+    pushBufferSize = conf.pushBufferMaxSize();
```
Comment (FMX) [inline client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java:121]: ditto.

File: common/src/main/scala/org/apache/celeborn/common/RssConf.scala [hunk 4]
```diff
@@ -364,13 +364,150 @@ class RssConf(loadDefaults: Boolean) extends Cloneable with Logging with Seriali
...
   def checkSlotsFinishedTimeoutMs: Long = get(WORKER_CHECK_SLOTS_FINISHED_TIMEOUT)
   def workerRecoverPath: String = get(WORKER_RECOVER_PATH)
   def partitionSorterCloseAwaitTimeMs: Long = get(PARTITION_SORTER_SHUTDOWN_TIMEOUT)
   def workerFlusherShutdownTimeoutMs: Long = get(WORKER_FLUSHER_SHUTDOWN_TIMEOUT)
+
+  // //////////////////////////////////////////////////////
+  //                      Flusher                       //
+  // //////////////////////////////////////////////////////
+  def workerFlusherBufferSize: Long = get(WORKER_FLUSHER_BUFFER_SIZE)
   def shuffleCommitTimeout: Long = get(WORKER_SHUFFLE_COMMIT_TIMEOUT)
```
Comment (FMX) [inline common/src/main/scala/org/apache/celeborn/common/RssConf.scala:511]: "shuffleCommitTimeout" can be moved to WORKER section.

**Review by FMX — COMMENTED** [review]
If a method name is renamed, maybe it's a good idea to  rename the corresponding variables as well?

File: common/src/main/scala/org/apache/celeborn/common/RssConf.scala (hunk 4 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/RssConf.scala]: DOne

File: client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java (hunk 1 above)
Comment (AngersZhuuuu) [reply client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java]: Done

File: client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java [hunk 5]
```diff
@@ -118,7 +118,7 @@ public SortBasedShuffleWriter(
     this.mapStatusRecords = new long[numPartitions];
     tmpRecords = new long[numPartitions];
 
-    pushBufferSize = RssConf.pushBufferMaxSize(conf);
+    pushBufferMaxSize = MaxMaxconf.pushBufferMaxSize();
```
Comment (FMX) [inline client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java:121]: typo?

File: client/src/main/java/org/apache/celeborn/client/read/RssInputStream.java [hunk 6]
```diff
@@ -138,9 +138,9 @@ private static final class RssInputStreamImpl extends RssInputStream {
       this.rangeReadFilter = RssConf.rangeReadFilterEnabled(conf);
 
       int headerLen = Decompressor.getCompressionHeaderLength(conf);
-      int blockSize = RssConf.pushBufferMaxSize(conf) + headerLen;
-      compressedBuf = new byte[blockSize];
-      decompressedBuf = new byte[blockSize];
+      int pushBufferMaxSize = conf.pushBufferMaxSize() + headerLen;
```
Comment (FMX) [inline client/src/main/java/org/apache/celeborn/client/read/RssInputStream.java:141]: It is actually not push buffer's max size.

File: common/src/main/scala/org/apache/celeborn/common/metrics/source/AbstractSource.scala [hunk 7]
```diff
@@ -38,21 +38,21 @@ case class NamedHistogram(name: String, histogram: Histogram)
...
+  val metricsSlidingWindowSize: Int = conf.metricsSlidingWindowSize
 
-  val sampleRate: Double = RssConf.metricsSampleRate(rssConf)
+  val metricsSampleRate: Double = conf.metricsSampleRate
 
-  val samplePerfCritical: Boolean = RssConf.metricsSamplePerfCritical(rssConf)
+  val samplePerfCritical: Boolean = RssConf.metricsSamplePerfCritical(conf)
 
-  final val InnerMetricsSize = RssConf.innerMetricsSize(rssConf)
+  final val InnerMetricsSize = RssConf.innerMetricsSize(conf)
```
Comment (FMX) [inline common/src/main/scala/org/apache/celeborn/common/metrics/source/AbstractSource.scala:51]: How about this?
```suggestion
  final val innerMetricsSize = RssConf.innerMetricsSize(conf)
```

File: client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java [hunk 8]
```diff
@@ -160,15 +160,15 @@ public <K, V> ShuffleWriter<K, V> getWriter(
         ShuffleClient client =
             ShuffleClient.get(
                 h.rssMetaServiceHost(), h.rssMetaServicePort(), rssConf, h.userIdentifier());
-        if ("sort".equals(RssConf.shuffleWriterMode(rssConf))) {
+        if ("sort".equals(rssConf.shuffleWriterMode())) {
```
Comment (FMX) [inline client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java:163]: rssconf -> conf ?

Comment (FMX) [reply client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java:163]: ok.

**Review by FMX — APPROVED** [review]
LGTM.

**Pull request merged**